Update tari script cucumber tests to use standalone miner

SWvheerden opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
This PR requires PR #2833 
This PR ports the cucumber tests to use the standalone miner and not the build in cucumber miner.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [ ] I'm merging against the `development` branch.
* [ ] I ran `cargo-fmt --all` before pushing.
* [ ] I ran `cargo test` successfully before submitting my PR.
* [ ] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

stringhandler commented on integration_tests/features/Propagation.feature:
```diff
@@ -4,8 +4,8 @@ Feature: Block Propagation
     And I have <NumNonSeeds> base nodes connected to all seed nodes
-    And I have a base node MINER connected to all seed nodes
-    When I mine <NumBlocks> blocks on MINER
+    And I have full mining package MINER connected to all seed nodes
```
This is not a good name. I am close to the source but I have no idea what this describes

SWvheerden replied:
I agree, but its the best I could come up with. 
This creates a: wallet called MINER, a node called MINER, and a standalone miner called: MINER.

delta1 replied:
How about "I have an integrated mining setup" or something like that

stringhandler replied:
Full mining package just sounds like it is inclusive of the merge mining proxy.

stringhandler replied:
I would suggest:
"When I have SHA3 miner MINER connected to all seed nodes"
"And I mine 5 blocks on MINER"

The non-named base node and wallet can be random strings

delta1 replied:
It also sounds like Kubrick movie 😆

![image](https://user-images.githubusercontent.com/351403/114711611-7660a100-9d2f-11eb-9cc3-5482e7e1f855.png)

stringhandler commented on integration_tests/features/Propagation.feature:
```diff
@@ -4,8 +4,8 @@ Feature: Block Propagation
-    And I have a base node MINER connected to all seed nodes
-    When I mine <NumBlocks> blocks on MINER
+    And I have full mining package MINER connected to all seed nodes
+    And Mining node MINER mines <NumBlocks> blocks
```
```suggestion
    And mining node MINER mines <NumBlocks> blocks
```

stringhandler commented on integration_tests/features/Propagation.feature:
```diff
@@ -60,34 +60,34 @@ Feature: Block Propagation
         # Wait for node to so start and get into listing mode
     When I wait 100 seconds
-    When I mine 5 blocks on MINER
+    Given Mining node MINER mines 5 blocks
```
This should be a `When`. `Given` is in the past tense and describes that something has been set up already. Either it should be `Given I have mined 5 blocks` or `When mining node MINER mines 5 blocks`

stringhandler commented:
Is it possible to retarget this to the development branch?

SWvheerden committed:
fix cucumber tests to use standalone miner

fix build of wallet_ffi

Update integration_tests/features/Propagation.feature

Co-authored-by: Mike the Tike <stringhandler@gmail.com>
update name 

fix naming

stringhandler commented on integration_tests/features/support/steps.js:
```diff
@@ -303,6 +303,87 @@ Given(
+  { timeout: 40 * 1000 },
+  async function (name, seed) {
+    //add the base_node
+    const node = this.createNode(name);
```
This duplication is not great, but I didn't see it in the first review. Hoping to add some clippy like stuff to our JS

stringhandler reviewed (approved):
Good enough to go in. Be careful of adding obviously duplicated code in future
